src/base/static/state/ducks/datasets.js

```diff
@@ -25,42 +25,62 @@ export function updateDatasetsLoadStatus(loadStatus) {
 }
 
 export function loadDatasets(datasets) {
-  const getTagChildren = (dataset, node) => {
-    const childNodes = [];
-
-    const traverse = (dataset, node) => {
-      if (node.parent) {
-        const newNode = dataset.tags.find(
-          childTag => childTag.id === node.parent,
-        );
-        childNodes.push(newNode);
-
-        traverse(dataset, newNode);
-      }
-    };
-    traverse(dataset, node);
+  const getTagDisplayName = (tag, tags) => {
+    const nodes = [];
+
+    let node = tag;
+    while (node.parent) {
+      nodes.push(node);
+      const parentNode = tags.find(tag => tag.id === node.parent);
+      node = parentNode;
+    }
+    nodes.push(node);
```

Comment:
I refactored this code because it caused some confusion - what we were calling `childNodes` here are actually "parent nodes".